src/shogun/base/AnyParameter.h

```diff
@@ -21,49 +27,102 @@ namespace shogun
 		GRADIENT_NOT_AVAILABLE = 0,
 		GRADIENT_AVAILABLE = 1
 	};
-
+	/** @brief Class AnyParameterProperties keeps track of parameter properties.
+	 *
+	 * Currently there are three types of parameters:
+	 * -# HYPERPARAMETER: the parameter determines how the training is
+	 * performed, e.g. regularisation
+	 * -# GRADIENT_PARAM: the parameter is used in gradient updates
+	 * -# MODEL_PARAM: model parameter that is trained, e.g. weight and bias
+	 * vectors and bias
+	 */
 	class AnyParameterProperties
 	{
 	public:
-		AnyParameterProperties()
-		    : m_description(), m_model_selection(MS_NOT_AVAILABLE),
-		      m_gradient(GRADIENT_NOT_AVAILABLE)
+		static const int32_t HYPERPARAMETER = 0x00000001;
```

Comment:
can we keep the naming self-consistent
HYPER_PARAM
or
HYPER_PARAMETER
or name the other one
MODELPARAMETER
or rather
MODEL_PARAMETER

Actually, in favour of short and concise names, why dont we use
HYPER
MODEL
GRADIENT
?
And then the constant itself can have a doxygen string to say what that means?

Reply:
OK! But hyperparameter is a term that could be misinterpreted if it written as HYPER, no?

Reply:
I think it is pretty clear since the class is ending with "Parameter", so everything in there is one ....

Reply:
The name MODEL causes conflicts if we use an enum, so will have to use something more specific, or just have these values as static class members of AnyParameterProperties.

Reply:
yes, please don't define them globally, but as a static member

Reply:
fine as well